testsuite/features/step_definitions/api_common.rb

```diff
@@ -245,21 +245,26 @@
   # Get the list of child channels for this base channel
   child_channels = $api_test.channel.software.list_child_channels(base_channel)
 
-  # Select all the child channels for this client
-  client.sub! 'ssh_minion', 'minion'
-  if client.include? 'buildhost'
-    selected_child_channels = ["custom_channel_#{client.sub('buildhost', 'minion')}", "custom_channel_#{client.sub('buildhost', 'client')}"]
-  elsif client.include? 'terminal'
-    selected_child_channels = ["custom_channel_#{client.sub('terminal', 'minion')}", "custom_channel_#{client.sub('terminal', 'client')}"]
-  else
-    custom_channel = "custom_channel_#{client}"
-    selected_child_channels = [custom_channel]
-  end
-  child_channels.each do |child_channel|
-    selected_child_channels.push(child_channel) unless child_channel.include? 'custom_channel'
-  end
+  # Filter out the custom channels
+  # This is needed because we might have both a traditional custom channel and a Salt custom channel
+  child_channels.reject! { |channel| channel.include? 'custom_channel' }
 
-  $api_test.activationkey.add_child_channels(key, selected_child_channels)
+  # Re-add the desired custom channel
+  client.sub! 'ssh_minion', 'minion'
+  client.sub! 'buildhost', 'minion'
+  client.sub! 'terminal', 'minion'
+  client.sub! 'monitoring_server', 'sle15sp4_minion'
+  custom_channel = if client.include? 'rocky8'
+                     "no-appstream-8-result-custom_channel_#{client}"
+                   elsif client.include? 'rocky9'
+                     "no-appstream-9-result-custom_channel_#{client}"
```

Comment:
I don't like to have defined here those channels.
Can't we use any map in the constants where these channels are already defined?

Reply:
We can, but that's yet another array for just one very specific case.

Should I do it nonetheless?

Reply:
i had a look, and no

the existing map is for the *base channels*, here we are dealing with the *custom chanels*

(`no-appstream-8-result-rhel8-pool-x86_64` versus `no-appstream-8-result-custom_channel_rocky8_minion`)

Reply:
OK, let's keep it for now, I don't want to slow down this task.
But if this grow, I think we need to move these definitions to a map in our constants, having definitions spread over the steps is bad in terms of maintenance.